src/circuit/v2/util.ts

```diff
@@ -0,0 +1,87 @@
+import { abortableDuplex, AbortError } from 'abortable-iterator'
+import { pipe } from 'it-pipe'
```

Comment:
`it-pipe` introduces a lot of latency as every operation in the pipeline crosses an async boundary even if the operation is synchronous.  This is likely to be a very hot code path so that latency will be expensive.